betelgeuse/__init__.py

```diff
@@ -352,7 +362,9 @@ def create_xml_testcase(config, testcase, automation_script_format):
             testcase.docstring = testcase.docstring.decode('utf8')
 
     # Check if any field needs a default value
-    testcase.fields = {k.lower(): v for k, v in testcase.fields.items()}
+    for k, v in testcase.fields.items():
+        testcase.fields.pop(k)
```

Comment:
In Python 2, `items()` returns copy of dict `key, value` pairs, but in Python 3 it returns special object that reflects changes in dict dynamically. Old code created new dict that we assigned to the same variable name as old one, now we modify dict in place. Iterating and modifying at the same time doesn't sound like a good idea to me, and Python 3 docs explicitly warns against it:

> Iterating views while adding or deleting entries in the dictionary may raise a RuntimeError or fail to iterate over all entries.

Reply:
I changed this. Now it creates new dictionary and then assigns it to the testcase.fields.